### PR TITLE
ignore emails that are from the reply by email addresses

### PR DESCRIPTION
**Pull request by ryantm**
discussion here: https://meta.discourse.org/t/spam-with-from-address-set-to-reply-by-email-address/87786/2

### CODE REVIEW

File: lib/email/receiver.rb [hunk 1]
```diff
@@ -107,6 +108,7 @@ def create_incoming_email
     def process_internal
       raise BouncedEmailError  if is_bounce?
       raise NoSenderDetectedError if @from_email.blank?
+      raise FromReplyByAddressError if is_from_reply_by_email_address?
```
Comment (ZogStriP) [inline lib/email/receiver.rb]: What happens when that exception is raised?

File: lib/email/receiver.rb [hunk 2]
```diff
@@ -203,6 +205,20 @@ def is_bounce?
       true
     end
 
+    def is_from_reply_by_email_address?
+      reply_addresses = [SiteSetting.reply_by_email_address]
```
Comment (ZogStriP) [inline lib/email/receiver.rb]: Why not use the `reply_by_email_address_regex` function?

File: lib/email/receiver.rb (hunk 1 above)
Comment (ryantm) [reply lib/email/receiver.rb]: I think it gets caught by https://github.com/discourse/discourse/blob/master/lib/email/processor.rb#L59 then handled by https://github.com/discourse/discourse/blob/master/lib/email/processor.rb#L71

File: lib/email/receiver.rb (hunk 2 above)
Comment (ryantm) [reply lib/email/receiver.rb]: reply_by_email_address_regex is too specific. I want to handle the case where the spammer has removed the + and the reply_key

File: lib/email/receiver.rb (hunk 1 above)
Comment (ryantm) [reply lib/email/receiver.rb]: Oh, I looked at that code more and it looks like it is probably still going to send a rejection mail in that case. I'll work on making it better.

File: lib/email/receiver.rb (hunk 2 above)
Comment (ZogStriP) [reply lib/email/receiver.rb]: Any chances you could merge both methods and add an argument?

**Comment by ryantm** [comment]
Okay, I think I made it actually work now! And I think I addressed @ZogStriP's concerns about copying the regexp function: I changed the regexp to not match everything if it was empty, and I match, even if the "+code" part is missing.

**Comment by ryantm** [comment]
ping @ZogStriP

**Comment by ZogStriP** [comment]
Thanks 👍

**Pull request merged**